open-api/rest-catalog-open-api.yaml

```diff
@@ -1435,12 +1463,102 @@ components:
           additionalProperties:
             type: string
 
+    OAuthTokenRequest:
+      description:
+        OAuth2 client credentials or token exchange request
+
+
+        See -
+
+        - Client credentials flow - https://datatracker.ietf.org/doc/html/rfc6749#section-4.4
+
+        - Token exchange flow - https://datatracker.ietf.org/doc/html/rfc8693
+      type: object
+      required:
+        - grant_type
+      properties:
+        grant_type:
+          type: string
+          enum:
+            - client_credentials
+            - urn:ietf:params:oauth:grant-type:token-exchange
+        client_id:
+          type: string
+          description:
+            Client ID for client credentials flow.
+
+
+            This can be sent in the request body, but OAuth2 recomments sending it in
+            a Basic Authorization header.
+        client_secret:
+          type: string
+          description:
+            Client secret for client credentials flow.
+
+
+            This can be sent in the request body, but OAuth2 recomments sending it in
+            a Basic Authorization header.
+        subject_token:
+          type: string
+          description: Subject token for token exchange request
+        subject_token_type:
+          type: string
+          description: Subject token type for token exchange request
+        scope:
+          type: string
 
   #############################
   # Reusable Response Objects #
   #############################
   responses:
 
+    OAuthTokenResponse:
```

Comment:
This doesn't quite match the [spec](https://datatracker.ietf.org/doc/html/rfc6749#section-5.1) for granting a token (it looks like it matches the token exchange response though)

Reply:
It's a superset of the fields in section 5.1 so that we can use it for both the client credentials and token exchange flows.

I just double-checked and it looks like the main problem is that `issued_token_type` is marked as required, even though it won't be present for client credentials. Is that what you were referring to? I'll make sure that only the fields that are required in both are actually required here. I'll also update each of these with a description so it is clear when they can or should be set.

Do you know when `token_type` and `issued_token_type` would both be used for token exchange? `issued_token_type` makes sense to me, but I'm not sure why `token_type` wouldn't simply be the one that was issued.

Reply:
Yeah, that's what I meant, that the required fields for the grant token response are only `access_token` and `token_type` in the spec. For token exchange, `issued_token_type` is also required. That's referring to the HTTP Authorization token type, e.g. most of the time set to `bearer`. Also, `expires_in` is recommended but optional (for both).

Reply:
Thanks for the clarification! I just assumed `issued_token_type` was the same thing as `token_type`, but that makes sense.